### PR TITLE
Add Exit Command

### LINKED ISSUE #302: Adapter_oM: Add execute command that exits the application
#### Description:

Add an `Execute` command `Exit` that closes the external software.

Needed for automated interop testing.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #302

Adding new command for fully closing down linked open instance of the software.

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EjpTgU6jo_tJr30rbJs0a_gBnZnhI1rYcv3aQ-Yc3A6E2g?e=fenJ0J

### CODE REVIEW

File: Adapter_oM/ExecuteCommands/Exit.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.Adapter.Commands
+{
+    [Description("Used to instruct the adapter to fully close down the running software. Adapter will generally not be useable after running this command.")]
+    public class Exit : IExecuteCommand
+    {
+        [Description("If set to true atempts to save any open file before exiting the software.")]
```
Comment (peterjamesnugent) [inline Adapter_oM/ExecuteCommands/Exit.cs:35]:
```suggestion
        [Description("If set to true, the adapter attempts to save any open file, closes the file and exits the software.")]
```

File: Adapter_oM/ExecuteCommands/Exit.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.oM.Adapter.Commands
+{
+    [Description("Used to instruct the adapter to fully close down the running software. Adapter will generally not be useable after running this command.")]
```
Comment (peterjamesnugent) [inline Adapter_oM/ExecuteCommands/Exit.cs:32]:
```suggestion
    [Description("Used to instruct the adapter to exit the software. The adapter will generally not be useable after running this command as the link to the software will be broken.")]
```
Do we explicitly say that this command is different to the `Close` command?

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Changes to descriptions.

**Review by peterjamesnugent — APPROVED** [review]
LGTM.

**Comment by peterjamesnugent** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 3408494013

This is being done in agreement with @IsakNaslundBh following discussion and is approved as a valid dispensation for this work.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 3409816604

Also done in discussion with @IsakNaslundBh to unblock this work for roll out of interop tests.

**Pull request merged**